src/intersection/compute_intersection.cc

```diff
@@ -281,54 +285,86 @@ bool ComputeIntersection< Simplex< 1  >, Simplex< 2  > >::compute_pathologic(uns
+{
+    ASSERT_EQ_DBG(0, IP12s.size());
     compute_plucker_products();
     computed_ = true;
+
+    //DebugOut() << "LINE: \n" << (*abscissa_)[0].point_coordinates()
+    //                       << (*abscissa_)[1].point_coordinates();
+
     
     // convert plucker products to local coords
     arma::vec3 w = {signed_plucker_product(0),
                     signed_plucker_product(1),
                     signed_plucker_product(2)};
     double w_sum = w[0] + w[1] + w[2];
     // if all products 0, we need avoiding zero division
-    if(std::abs(w_sum) > rounding_epsilonX)
+    //DebugOut().VarFmt(w_sum);
+
+    if(std::abs(w_sum) > rounding_epsilon)
         w = w / w_sum;
     
-    // test whether all plucker products have the same sign
-    if(((w[0] > rounding_epsilon) && (w[1] > rounding_epsilon) && (w[2] > rounding_epsilon)) ||
-       ((w[0] < -rounding_epsilon) && (w[1] < -rounding_epsilon) && (w[2] < -rounding_epsilon))){
-        
+    unsigned int n_positive = 0;
+    unsigned int n_negative = 0;
+    unsigned int zero_idx_sum =0;
+
+    for (unsigned int i=0; i < 3; i++) {
+        //DebugOut().VarFmt(i).VarFmt(w[i]);
+        if (w[i] > rounding_epsilon) n_positive++;
+        else if ( w[i] > -rounding_epsilon) zero_idx_sum+=i;
+        else n_negative++;
+    }
+
+    // any negative barycentric coordinate means, no intersection
+    if (n_negative>0) return IntersectionResult::none;
+
+    // test whether any plucker products is non-zero
+    if (n_positive > 0) {
         IntersectionPointAux<1,2> IP;
         
-        if(compute_plucker(IP, w))
-        {
-            IP12s.push_back(IP);
-            return true;
+        compute_plucker(IP, w);
+        // edge of triangle
+        unsigned int non_zero_idx=0;
+        if (n_positive == 2) {
+            // one zero product, intersection on the zero edge
+            // the zero edge index is equal to zero_idx_sum
+            IP.set_topology_B(zero_idx_sum, 1);
+            non_zero_idx =  (zero_idx_sum + 1) % 3;
         }
-    }else if(compute_zeros_plucker_products){
-        
-        // 1 zero product -> IP is on the triangle side
-        // 2 zero products -> IP is at the vertex of triangle (there is no other IP)
+        else if (n_positive == 1) {
+            // two zero products, intersection in vertex oposite to single non-zero edge
+            // index of the non-zero edge is 3-zero_idx_sum
+            IP.set_topology_B(RefElement<2>::oposite_node(3-zero_idx_sum), 0);
+            non_zero_idx = 3-zero_idx_sum;
+        }
+        //DebugOut().VarFmt(non_zero_idx).VarFmt(signed_plucker_product(non_zero_idx));
+
+        auto result = signed_plucker_product(non_zero_idx) > 0 ?
+                IntersectionResult::positive : IntersectionResult::negative;
+        IP.set_orientation(result);
+
+        IP12s.push_back(IP);
+        return result;
+
+    } else if(compute_zeros_plucker_products){
```

Comment:
so the flag "compute_zeros_plucker_products" is not needed anymore, correct?
at the end, we have only compute and compute_final with same argument..